File: ios/Runner/BikeStationStatus.swift
//
//  Bikes.swift
//  Runner
//
//  Created by Davenson Lombard on 2020-02-25.
//  Copyright © 2020 The Chromium Authors. All rights reserved.
//

import Foundation
import RealmSwift

class BikeStationStatus: Object {
    @objc dynamic var station_id: String = ""
    @objc dynamic var num_bikes_available: Int = 0
    @objc dynamic var num_ebikes_available: Int = 0
    @objc dynamic var num_bikes_disabled: Int = 0
    @objc dynamic var num_docks_available: Int = 0
    @objc dynamic var num_docks_disabled: Int = 0
    @objc dynamic var is_installed: Int = 0
    @objc dynamic var is_renting: Int = 0
    @objc dynamic var is_returning: Int = 0
    @objc dynamic var last_reported: Int = 0

    override static func primaryKey() -> String? {
        return "station_id"
    }
}

File: lib/models/bike_station_status.dart
import 'package:json_annotation/json_annotation.dart';
part 'bike_station_status.g.dart';

@JsonSerializable()
class BikeStationStatus{
  String station_id;
  int num_bikes_available;
  int num_ebikes_available;
  int num_bikes_disabled;
  int num_docks_available;
  int num_docks_disabled;
  int is_installed;
  int is_renting;
  int is_returning;
  int last_reported;

  BikeStationStatus(
      this.station_id,
      this.num_bikes_available,
      this.num_bikes_disabled,
      this.num_docks_available,
      this.num_docks_disabled,
      this.num_ebikes_available,
      this.is_installed,
      this.is_renting,
      this.is_returning,
      this.last_reported);

  factory BikeStationStatus.fromJson(Map<String, dynamic> json) =>
      _$BikeStationStatusFromJson(json);

  Map<String, dynamic> toJson() => _$BikeStationStatusToJson(this);
}
